homeassistant/components/toon/climate.py

```diff
@@ -1,90 +1,153 @@
-    STATE_COOL, STATE_ECO, STATE_HEAT, STATE_AUTO,
+    STATE_AUTO, STATE_COOL, STATE_DRY, STATE_ECO, STATE_HEAT,
     SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
-import homeassistant.components.toon as toon_main
+from homeassistant.components.toon import ToonEntity
+from homeassistant.components.toon.const import DATA_TOON_CLIENT, DOMAIN
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.helpers.typing import HomeAssistantType
+
+DEPENDENCIES = ['toon']
+
+_LOGGER = logging.getLogger(__name__)
 
 SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
 
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+
 HA_TOON = {
     STATE_AUTO: 'Comfort',
     STATE_HEAT: 'Home',
     STATE_ECO: 'Away',
     STATE_COOL: 'Sleep',
 }
+
 TOON_HA = {value: key for key, value in HA_TOON.items()}
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Toon climate device."""
-    add_entities([ThermostatDevice(hass)], True)
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry,
+                            async_add_entities) -> None:
+    """Set up a Toon binary sensors based on a config entry."""
+    toon = hass.data[DATA_TOON_CLIENT][entry.entry_id]
+    async_add_entities([ToonThermostatDevice(toon)], True)
 
 
-class ThermostatDevice(ClimateDevice):
+class ToonThermostatDevice(ToonEntity, ClimateDevice):
     """Representation of a Toon climate device."""
 
-    def __init__(self, hass):
+    def __init__(self, toon) -> None:
         """Initialize the Toon climate device."""
-        self._name = 'Toon van Eneco'
-        self.hass = hass
-        self.thermos = hass.data[toon_main.TOON_HANDLE]
-
         self._state = None
-        self._temperature = None
-        self._setpoint = None
-        self._operation_list = [
-            STATE_AUTO,
-            STATE_HEAT,
-            STATE_ECO,
-            STATE_COOL,
-        ]
+
+        self._current_temperature = None
+        self._target_temperature = None
+        self._real_target_temperature = None
+        self._next_target_temperature = None
+        self._modulation_level = None
+
+        self._heating_type = None
+        self._program_state = None
+        self._program_next = None
+        self._holiday_state = None
+
+        super().__init__(toon, "Toon Thermostat", 'mdi:thermostat')
 
     @property
-    def supported_features(self):
-        """Return the list of supported features."""
-        return SUPPORT_FLAGS
+    def unique_id(self) -> str:
+        """Return the unique ID for this thermostat."""
+        return self.toon.agreement.id
 
     @property
-    def name(self):
-        """Return the name of this thermostat."""
-        return self._name
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information about this thermostat."""
+        toon = self.toon.agreement
```

Comment:
That's a bit of a confusing variable name, I would have expected `agreement` oid

Reply:
Yep, thanks 👍 , addressed.